Game/Game/Game/Game.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace Game;
+
+public class Game
+{
+    private readonly Action<int, int> action;
+    private int currentPositionOnX;
+    private int currentPositionOnY;
+    private readonly string[] map;
+
+    public Game(int initialPositionOnX, int initialPositionOnY, string pathToFile, Action <int, int> action)
+    {
+        currentPositionOnX = initialPositionOnX;
+        currentPositionOnY = initialPositionOnY;
+        map = File.ReadAllLines(pathToFile);
+        this.action = action;
+        PrintMap(this.map);
+        action(currentPositionOnX, currentPositionOnY);
+        DrawCharacter();
+    }
+
+
+    private static void DrawCharacter()
+    {
+        Console.WriteLine("@");
+    }
+
+    private static void PrintMap(string[] map)
+    {
+        for (int i = 0; i < map.Length; i++)
+        {
+            for (int j = 0; j < map[i].Length; j++)
+            {
+                Console.Write(map[i][j]);
+            }
+            Console.WriteLine();
+        }
+    }
+
+    private static bool IsWall(char x) => x == '|' || x == '+' || x == '-' || x == '_';
+
+    public void Move(Func<int, int, (int, int)> func)
+    {
+        var (newPositionOnX, newPositionOnY) = func(currentPositionOnX, currentPositionOnY);
+        action(newPositionOnX, newPositionOnY);
+        if (IsWall(map[newPositionOnY][newPositionOnX]))
+        {
+            action(currentPositionOnX, currentPositionOnY);
+            return;
+        }
+
+        Console.Write("@");
+        action(currentPositionOnX, currentPositionOnY);
+        Console.WriteLine(" ");
+        action(newPositionOnX, newPositionOnY);
+        (currentPositionOnX, currentPositionOnY) = (newPositionOnX, newPositionOnY);
+    }
+
+    public void OnLeft(object? sender, EventArgs args)
+    {
+        Move((x, y) => (x - 1, y));
+    }
```

Comment:
```suggestion
    public void OnLeft(object? sender, EventArgs args)
        => Move((x, y) => (x - 1, y));
```
Это просто делегация, так что даже по смыслу надо писать через `=>`